client/python/test/test_client.py

```diff
@@ -21,6 +22,23 @@
 HF_TOKEN = "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"  # Intentionally revealing this key for testing purposes
 
 
+@contextmanager
+def connect(demo: gr.Blocks):
+    _, local_url, _ = demo.launch(prevent_thread_lock=True)
+    try:
+        yield Client(local_url)
+    finally:
+        # A more verbose version of .close()
+        # because we should set a timeout
+        # the tests that call .cancel() can get stuck
+        # waiting for the thread to join
```

Comment:
Would it be better to add an optional `timeout` parameter to `close()` then?

Reply:
Yes not a bad idea but the client tests use the latest version of gradio so it would not be available for this pr.